apps/linkedin-to-notion/src/tabs/onboarding.tsx

```diff
@@ -0,0 +1,169 @@
+    current: 'default',
+    next: 'disabled',
+    done: 'done',
+  };
+
+  const checkIsPinned = async (): Promise<void> => {
+    const userSettings = await chrome.action.getUserSettings();
+
+    if (userSettings.isOnToolbar === true) {
+      const userData = await sendToBackground({
+        name: 'users/resolvers/updateOnboardingStatus',
+        body: { id: user.id, onboardingStatus: OnboardingStatus.EXTENSION_PINNED },
+      });
+      setUser(userData);
+      return;
+    }
+
+    alert('Recruitivity does not seem to be pinned. Please pin it and try again.');
+  };
+
+  return (
+    <BasicLayout>
+      {numberOfTasks === 0 && <Confetti />}
+      <Heading1 color="plasmo-text-white" className="plasmo-text-center plasmo-px-16 plasmo-w-5/6">
+        {headingContent[numberOfTasks]}
+      </Heading1>
+      <div className="plasmo-grid plasmo-grid-cols-1 md:plasmo-grid-cols-2 xl:plasmo-grid-cols-3 plasmo-gap-14 plasmo-items-center">
+        <Card
+          className="plasmo-bg-background-light"
+          title={<Heading2>1 - Connect to Notion</Heading2>}
+          icon={<img src={notion_logo} alt="Notion Logo" width={150} height={150} />}
+          callToAction={
+            <ButtonPrimary
+              state={cardStateToButtonState[state.connectToNotion]}
+              onClick={async () =>
+                await handleOAuthLogin('notion', {
+                  redirectUrl: routes.tabs.onboarding,
+                })
+              }>
+              {cardStateToButtonState[state.connectToNotion] === 'done' ? 'Done' : 'Connect to Notion'}
+            </ButtonPrimary>
+          }
+          state={state.connectToNotion}
+        />
+        <Card
+          className="plasmo-bg-background-light"
+          title={<Heading2>2 - Save a first profile</Heading2>}
+          icon={<img src={linkedin_logo} alt="LinkedIn Logo" width={150} height={150} />}
+          callToAction={
+            <ButtonPrimary
+              state={cardStateToButtonState[state.saveFirstProfile]}
+              onClick={() => {
+                window.location.href = routes.linkedin.me;
+              }}>
+              {cardStateToButtonState[state.saveFirstProfile] === 'done' ? 'Done' : 'Go to LinkedIn'}
+            </ButtonPrimary>
+          }
+          state={state.saveFirstProfile}
+        />
+        {state.pinExtension === 'current' && (
+          <img
+            className="plasmo-hidden xl:plasmo-block plasmo-absolute plasmo-top-6 plasmo-right-24 -plasmo-z-10"
+            src={arrow}
+            alt="Pin Icon"
+          />
+        )}
+        <Card
+          className="plasmo-bg-background-light"
+          title={<Heading2>3 - Pin the extension</Heading2>}
+          icon={<img src={pin_icon} alt="Pin Icon" width={150} height={150} />}
+          state={state.pinExtension}
+          callToAction={
+            <ButtonPrimary state={cardStateToButtonState[state.pinExtension]} onClick={checkIsPinned}>
```

Comment:
Following up on our conversion about checking whether or not the extension is pinned, I think that once the user has reached the second step of the onboarding, we could check if the extension has been pinned every second. It's not intensive operation and it would be much less confusing imo. Wdyt?

Reply:
I don't think so, I find it a bit confusing to have stuff happening for mysterious reason. Adding the validate button makes it very straight forward

Reply:
I disagree, especially since the behavior you originally expected due to my bad explanation of `chrome.userSettings`. We explicitly ask the user to pin the extension. If one second after pinning it, confettis burst on screen, I think it's straight forward that it's the extension pinning that caused it. No?